### DISCUSSION TITLE
How to export/import a JupyterLite between browsers?

### DISCUSSION THREAD

**Discussion by deeplook** [jupyterlite/jupyterlite#437] [Q&A]
I'll need to switch to a new laptop, soon, and try to understand how I would export an existing JupyterLite "installation" from one browser to the new machine? Or even simpler, from one browser application to another, like Firefox to Safari, say. Is there any procedure to achieve this already?

**Comment by bollwyvl** [comment]
At present, there is no real way to do this en masse, but this is a good idea, and the "happy path" of #9 and #407. Additionally, it would influence #315, as different storage systems might not support this.  

At present, the only work-arounds i can think of are:

- drag and drop all the files by hand
- write an import/export mechanism yourself inside a kernel against the browser api (alas, raw `IndexedDB` is _not_ pretty).

At the core of it, the things we store in IndexedDB are:

- settings
- contents

I'm somewhat on the fence as to whether this would be a built-in or an extension... especially if it required some additional libraries. Presumably, we could write something that could scrape these out and store them in some intermediate format, which could then be read back out and restored. 

The settings already have a canonical format (`overrides.json`). Content... not so much. To maintain all the data, it could be a strict JSON dump of the Jupyter Contents API (slightly different than our storage format), but it's not really a very good archival format, because nothing speaks it. 

Perhaps better and more portable would be a `.tar.gz` file that encoded all of the details into the filesystem.

A bridge too far might be if we had no-fooling `git` or `fossil` in the browser, and could actually export something with history.

Indeed, there's the dream, someday, that one might be able to build actual, portable _sites_ (e.g. with new extensions) inside the browser, but there are a number of limitations, including fairly iron-clad browser security ones, that would make this hard.
